lib/elixir/lib/file/stream.ex

```diff
@@ -132,9 +136,20 @@ defmodule File.Stream do
       {:error, __MODULE__}
     end
 
-    defp count_raw_bom(path, modes) do
+    defp open!(stream, modes, fun) do
+      with {:ok, device} <- File.Stream.__open__(stream, read_modes(modes)),
+           res = fun.(device),
+           :ok <- :file.close(device) do
```

Comment:
If my understanding is correct, the originally used `File.open/3` was doing a [`try/after`](https://github.com/elixir-lang/elixir/blob/main/lib/elixir/lib/file.ex#L1484-L1488) to make sure we are closing when `fun` fails, so this might not be strictly equivalent. Is it intentional / fine in this case?

Reply:
Good catch.